### ISSUE TITLE
Tracking stops under specific conditions 

### ISSUE THREAD

**Issue by keynlotepan** [herverenault/Self-Hosted-GPS-Tracker#35]
During my yesterday's run, Self-Hosted-GPS-Tracker stopped tracking while I was using Firefox browser concurrently. The first time this happened I was not sure whether it was result of a false operation, but yesterday I did pay attention, so I am sure that it stopped by itself. When I fetched Self-Hosted-GPS-Tracker back to foreground, it saw that it was in switched off state. According to the [state diagram](https://www.thenewboston.com/forum/topic.php?id=3234) I assume that Firefox consumed much memory and thus Self-Hosted-GPS-Tracker was killed.

According to [this post](http://stackoverflow.com/questions/9434179/how-to-keep-a-service-running-in-background-even-after-user-quits-the-app) I think Self-Hosted-GPS-Tracker or parts of it should be implemented as a [Service](http://developer.android.com/guide/components/services.html)

It would not be enough to store the "on" state and restore it when it comes back to foreground because during it's time in background it would stop and not start tracking before the user fetches it into foreground again.

**Comment by herverenault** [comment]
I've never seen this happen although I've used Firefox + OSMAnd (which both eat a lot of memory) and this app is already implemented as a [foreground Service](https://developer.android.com/reference/android/app/Service.html#startForeground(int,%20android.app.Notification)), so it should not be killed. But then...

Maybe I could call [setIntentRedelivery(true)](https://developer.android.com/reference/android/app/IntentService.html#setIntentRedelivery%28boolean%29) to see if it helps.

**Comment by herverenault** [comment]
How can we try that? Do you build the app yourself? Or do you get it from a store? (Google Play or F-Droid)

**Comment by keynlotepan** [comment]
I am using current version of your App as delivered from F-Droid (my device is operated without a Google account). Thanks for

**Comment by keynlotepan** [comment]
_"Maybe I could call setIntentRedelivery(true) to see if it helps."_
But in this case the state of the app (tracking off or on) must be stored on every change and must be re-established entering onCreate() to continue tracking. This could have the side-effect to irritate users switching off their phones (some users do this at night) without switching off tracking before not expecting that the app would restart in "on" state next time.

**Comment by herverenault** [comment]
> ... state of the app (tracking off or on) must be stored ...
Yes, indeed.

> ... irritate users switching off their phones without ...
This could be a pref, don't you think? Something like "Restart tracking if phone is rebooted: Yes/No"

**Comment by keynlotepan** [comment]
> This could be a pref, don't you think? Something like "Restart tracking if phone is rebooted: Yes/No"
I did not dare to propose this, not having the impression that you are interested to overload your app with options. I think "Resume tracking..." describes the case that tracking was tuned on before more precisely, but my English skills are not very presentable...

**Comment by herverenault** [comment]
You're right: I want to keep it simple. But I would like to add important features like the one you suggest. Problem is I don't have time for this right now... I hope things improve in coming weeks.
